src/NLog/Config/ServiceRepository.cs

```diff
@@ -62,7 +65,7 @@ internal ServiceRepository(bool resetGlobalCache = false)
                 ConfigurationItemFactory.Default = null;    //build new global factory
 
             this.RegisterDefaults();
-            CreateInstance = DefaultResolveInstance;
+            CreateInstance = itemType => DefaultResolveInstance(itemType, new HashSet<Type>());
```

Comment:
Please don't allocate upfront.

Reply:
fixed

(PS: I will always try to fix it everywhere)